go/vt/tabletserver/gorpcqueryservice/sqlquery.go

```diff
@@ -23,13 +23,40 @@ type SqlQuery struct {
 // GetSessionId is exposing tabletserver.SqlQuery.GetSessionId
 func (sq *SqlQuery) GetSessionId(sessionParams *proto.SessionParams, sessionInfo *proto.SessionInfo) (err error) {
 	defer sq.server.HandlePanic(&err)
-	return sq.server.GetSessionId(sessionParams, sessionInfo)
+	tErr := sq.server.GetSessionId(sessionParams, sessionInfo)
+	tabletserver.AddTabletErrorToSessionInfo(tErr, sessionInfo)
+	if *tabletserver.RPCErrorOnlyInReply {
+		return nil
+	}
+	return tErr
 }
 
 // Begin is exposing tabletserver.SqlQuery.Begin
 func (sq *SqlQuery) Begin(ctx context.Context, session *proto.Session, txInfo *proto.TransactionInfo) (err error) {
 	defer sq.server.HandlePanic(&err)
-	return sq.server.Begin(callinfo.RPCWrapCallInfo(ctx), session, txInfo)
+	tErr := sq.server.Begin(callinfo.RPCWrapCallInfo(ctx), session, txInfo)
+	tabletserver.AddTabletErrorToTransactionInfo(tErr, txInfo)
+	if *tabletserver.RPCErrorOnlyInReply {
+		return nil
+	}
+	return tErr
+}
+
+// UnsupportedNewCommit should not be used by anything other than tests.
+// It will eventually replace Commit, but it breaks compatibility with older clients.
+// Once all clients are upgraded, it can be replaced.
+func (sq *SqlQuery) UnsupportedNewCommit(ctx context.Context, commitRequest *proto.CommitRequest, commitResponse *proto.CommitResponse) (err error) {
```

Comment:
We can do a 2-step migration to the new RPCs (begin, commit, rollback).
It is better to name them Begin2/Commit2/Rollback2